### ISSUE TITLE
Birdtray hard codes a default path

### ISSUE THREAD

**Issue by Syonis** [NixOS/nixpkgs#291745]
Birdtray hard codes a default path

1. nix-shell -p birdtray
2. start birdtray

### Expected behavior
If you leave it as it is Birdtray tries to start Thunderbird with "/usr/bin/thunderbird"
If you change that to "thunderbird" and launch birdtray it tries to launch "home/<user>/thunderbird"

### Screenshots
<img width="253" alt="birdtray_1" src="https://github.com/NixOS/nixpkgs/assets/97046583/753473cb-addf-4a5b-a36f-1ffb1870542c">
<img width="713" alt="birdtray_2" src="https://github.com/NixOS/nixpkgs/assets/97046583/1249ec16-e2a6-463e-822b-8763bcc8b4f4">

### Notify maintainers
@Flakebi 

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by eclairevoyant** [comment]
If you've added thunderbird to `environment.systemPackages`, then something like `/run/current-system/sw/bin/thunderbird` should work in the settings.

**Comment by Syonis** [comment]
Jep, thx. that worked.

**Comment by AndrewKvalheim** [comment]
Ideally the package should manage this e.g. via [`OPT_THUNDERBIRD_CMDLINE`](https://github.com/gyunaev/birdtray/blob/7261ff0c139b98cf7cf9a1c8727ab74dee82756d/src/utils.cpp#L199).

**Comment by eclairevoyant** [comment]
That's only going to set the _default_, and will become wrong if the thunderbird storepath changes.

**Comment by AndrewKvalheim** [comment]
I see, you’re pointing out that the default is *copied* to a stateful configuration file rather than merely used as a fallback.

**Comment by eclairevoyant** [comment]
Yes, and the copying only happens if the config file doesn't already exist. I'm inclined to say that upstream should consider reading from PATH, or provide the capability to do so.

**Comment by dotlambda** [comment]
Why don't we patch https://github.com/gyunaev/birdtray/blob/00110c50398cbae4d6da4521ccbf5d7166eac670/src/utils.cpp#L207? Replace the path with `lib.getExe pkgs.thunderbird`.

**Comment by eclairevoyant** [comment]
As said above, that will only affect the _default_ path to thunderbird. That would break as soon as thunderbird updates and the config file points to a nonexistent binary.

**Comment by Sikeen** [comment]
what about a check for whether the system is nix or not? would that work?

**Comment by Sikeen** [comment]
in this case, another if statement in the code

**Comment by eclairevoyant** [comment]
Now that upstream has made a fix, I've created a PR to pull in that commit. Please give the PR #327363 a test and let me know if it works as expected.

**Comment by eclairevoyant** [comment]
Fair, I don't really use birdtray, so I don't know how it works, just made a best-effort attempt.
If it doesn't work as designed, I can close it out and leave it for someone else to work on.

**Comment by OscarCunningham** [comment]
@ddogfoodd I just got birdtray working on NixOS myself. I had the issue in this thread, which I fixed by changing `/usr/bin/thunderbird` to `/run/current-system/sw/bin/thunderbird` in my birdtray settings. But I also had this issue https://github.com/gyunaev/birdtray/issues/426 with an incompatibility between birdtray and wayland. I started birdtray with the `WAYLAND_DISPLAY` environment variable disabled, and it worked perfectly.

The reason I mention this is that it sounds like the fix above might actually have worked, but it still seems buggy to you because you're also experiencing the wayland bug.

**Comment by eclairevoyant** [comment]
Unsubscribing as I don't use birdtray.

**Comment by BlueDrink9** [comment]
How come it couldn't be patched just by setting `OPT_THUNDERBIRD_CMDLINE=/run/current-system/sw/bin/thunderbird`? Then it will still work across changes in Thunderbird binary
